lib/frr_pthread.c

```diff
@@ -107,6 +107,9 @@ struct frr_pthread *frr_pthread_new(struct frr_pthread_attr *attr,
 			fpt->attr = *attr;
 			name = (name ? name : "Anonymous thread");
 			fpt->name = XSTRDUP(MTYPE_FRR_PTHREAD, name);
+			memset(fpt->os_name, '\0', THRD_NAMELEN);
+			if (os_name)
+				strncpy(fpt->os_name, os_name, THRD_NAMELEN-1);
```

Comment:
I would drop the `memset` and use a string function that is guaranteed to null terminate, a la
```
if (os_name)
    snprintf(fpt->os_name, THRD_NAMELEN - 1, "%s", os_name);
```